### PR TITLE
Update DG with newly added Group model

### PR DESCRIPTION
**Pull request by tohlh**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -433,6 +433,29 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+**Extensions**
+
+* 1a. The addgroup command parameters are invalid or incomplete.
+
+    * 1a1. TutorsContactsPro shows an error message.
+
+      Use case resumes at step 2.
+
+* 1b. TutorsContactsPro detects that the student already exists on the list.
```
Comment (jocelyn-soh) [inline docs/DeveloperGuide.md:453]: I think you meant "group" instead of "student"?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -433,6 +433,29 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+    * 1b1. TutorsContactsPro informs the tutor that the group already exists on the list.
+    * 1b2. Tutor confirms cancellation of adding the group.
+
+      Use case ends.
```
Comment (jocelyn-soh) [inline docs/DeveloperGuide.md:458]: I like that the extensions are comprehensive, good job!

**Review by jocelyn-soh — COMMENTED** [review]
LGTM, except for a small typo.

**Review by STELLA-LYE — APPROVED** [review]
LGTM.

**Pull request merged**